src/basic/strv.c

```diff
@@ -353,6 +353,58 @@ int strv_split_extract(char ***t, const char *s, const char *separators, Extract
         return (int) n;
 }
 
+int strv_split_extract_pairs_colon(char ***t, const char *s) {
+        _cleanup_strv_free_ char **l = NULL;
+        size_t n = 0, allocated = 0;
+        int r;
+
+        assert(t);
+        assert(s);
+
+        for (;;) {
+                _cleanup_free_ char *first = NULL, *second = NULL, *tuple = NULL, *second_or_empty = NULL;
+                const char *p = NULL;
+
+                r = extract_first_word(&s, &tuple, WHITESPACE, EXTRACT_CUNESCAPE|EXTRACT_CUNESCAPE_RELAX|EXTRACT_UNQUOTE|EXTRACT_DONT_COALESCE_SEPARATORS);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        break;
+
+                p = tuple;
+                r = extract_many_words(&p, ":", EXTRACT_CUNESCAPE_RELAX|EXTRACT_UNQUOTE|EXTRACT_DONT_COALESCE_SEPARATORS, &first, &second, NULL);
```

Comment:
All three flags seem wrong here. Same reasons as above, and in addition EXTRACT_UNQUOTE would mean a second level of unquoting. I don't think we want that.

Reply:
EXTRACT_CUNESCAPE_RELAX is needed, otherwise ```"ten\\:eleven\\\\"``` returns EINVAL